src/ESPEasy-Globals.h

```diff
@@ -340,17 +340,24 @@ void check_size() {
 
 #define CPLUGIN_WEBFORM_SHOW_CONFIG        57
 
-#define CONTROLLER_HOSTNAME                 1
-#define CONTROLLER_IP                       2
-#define CONTROLLER_PORT                     3
-#define CONTROLLER_USER                     4
-#define CONTROLLER_PASS                     5
-#define CONTROLLER_SUBSCRIBE                6
-#define CONTROLLER_PUBLISH                  7
-#define CONTROLLER_LWT_TOPIC                8
-#define CONTROLLER_LWT_CONNECT_MESSAGE      9
-#define CONTROLLER_LWT_DISCONNECT_MESSAGE  10
-#define CONTROLLER_TIMEOUT                 11
+#define CONTROLLER_USE_DNS                  1
+#define CONTROLLER_HOSTNAME                 2
+#define CONTROLLER_IP                       3 
+#define CONTROLLER_PORT                     4
+#define CONTROLLER_USER                     5
+#define CONTROLLER_PASS                     6
+#define CONTROLLER_MIN_SEND_INTERVAL        7
+#define CONTROLLER_MAX_QUEUE_DEPTH          8
+#define CONTROLLER_MAX_RETRIES              9
+#define CONTROLLER_FULL_QUEUE_ACTION        10
+#define CONTROLLER_CHECK_REPLY              12
+#define CONTROLLER_SUBSCRIBE                13
+#define CONTROLLER_PUBLISH                  14
+#define CONTROLLER_LWT_TOPIC                15
+#define CONTROLLER_LWT_CONNECT_MESSAGE      16
+#define CONTROLLER_LWT_DISCONNECT_MESSAGE   17
+#define CONTROLLER_TIMEOUT                  18
+#define CONTROLLER_ENABLED                  19  // Keep this as last, is used to loop over all parameters
```

Comment:
what about enum here?

Reply:
Yeah, I guess you're right, then the compiler also can warn if you forget some in a switch statement